app/src/main/java/io/appwrite/almostnetflix/feed/MovieDataSource.kt

```diff
@@ -56,27 +65,31 @@ class MovieDataSource(
     }
 
     suspend fun getMyWatchlist(userId: String, movieIds: List<String>): List<Movie> {
-        val movies = database.listDocuments(
+        val movies = databases.listDocuments(
+            databaseId = Configuration.DATABASE_ID,
             collectionId = Configuration.MOVIE_COLLECTION_ID,
             queries = listOf(
-                Query.equal("\$id", movieIds)
+                Query.equal("\$id", value = movieIds.toString()),
```

Comment:
The value should be the entire array here:

```suggestion
                Query.equal("\$id", value = movieIds),
```